src/platform/common/utils/multiStepInput.ts

```diff
@@ -266,6 +266,9 @@ export class MultiStepInput<S> implements IMultiStepInput<S> {
                             resolve(inputValue);
                         } else {
                             input.validationMessage = validationMessage;
+
+                            // On validation error make sure we are showing our input
+                            input.show();
```

Comment:
Smaller fix here, with the new validation flow the main validation all happens with the server selection multistep input. But when coming back from the failed password validation the server selection input was not actually getting shown again (this was exposed by fixing the bug again). Make sure we show our input again so the customer can see the password error.